### PR TITLE
MGMT-16926: fix validation bug + allow CNV and LVM multi node

### PR DESCRIPTION
**Pull request by eifrach**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/operators/lvm/lvm_operator.go [hunk 1]
```diff
@@ -104,19 +104,19 @@ func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, ho
...
 	message := "Logical Volume Manager requires at least one non-installation HDD/SSD disk on the host"
-	// if (role == models.HostRoleWorker && !*cluster.SchedulableMasters) || *cluster.SchedulableMasters {
-	if role == models.HostRoleWorker || *cluster.SchedulableMasters {
+	if role == models.HostRoleWorker || areSchedulable {
 		if diskCount == 0 {
 			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{message}}, nil
 		}
+	} else {
+		if role == models.HostRoleAutoAssign {
+			status := "For LVM Standard Mode, host role must be assigned to master or worker."
```
Comment (gamli75) [inline internal/operators/lvm/lvm_operator.go:116]:
```suggestion
			status := "For Logical Volume Manager Standard Mode, host role must be assigned to master or worker."
```

File: internal/operators/manager.go [hunk 2]
```diff
@@ -463,13 +448,25 @@ func EnsureLVMAndCNVDoNotClash(cluster *common.Cluster, openshiftVersion string,
 		}
 	}
 
-	if lvmEnabled && cnvEnabled {
+	if !lvmEnabled || !cnvEnabled {
+		return nil
+	}
+
+	// Openshift version greater or Equal to 4.12.0 support cnv and lvms
+	if isGreaterOrEqual, _ := common.BaseVersionGreaterOrEqual(lvm.LvmsMinOpenshiftVersion4_12, openshiftVersion); !isGreaterOrEqual {
```
Comment (gamli75) [inline internal/operators/manager.go:456]: there is a mix(ODF + LVM) in the error message - not your code but please fix it
should be "return errors.Errorf("Currently, you can not install Logical Volume Manager operator at the same time as Virtualization operator.")"

File: internal/operators/manager.go [hunk 3]
```diff
@@ -463,13 +448,25 @@ func EnsureLVMAndCNVDoNotClash(cluster *common.Cluster, openshiftVersion string,
...
 		return errors.Errorf("Currently, you can not install OpenShift Data Foundation Logical Volume Manager operator at the same time as Virtualization operator.")
 	}
 
-	if lvmEnabled && !common.IsSingleNodeCluster(cluster) {
-		return errors.Errorf("OpenShift Data Foundation Logical Volume Manager operator is only supported for Single Node Openshift")
+	if isGreaterOrEqual, _ := common.BaseVersionGreaterOrEqual(lvm.LvmMinMultiNodeSupportVersion, openshiftVersion); isGreaterOrEqual {
+		return nil
+	}
+
+	// before 4.15 multi node LVM not supported
```
Comment (gamli75) [inline internal/operators/manager.go:464]: I think that this condition should be before 
`if isGreaterOrEqual, _ := common.BaseVersionGreaterOrEqual(lvm.LvmMinMultiNodeSupportVersion, openshiftVersion); isGreaterOrEqual {`
let's see if your unit test will pass

File: internal/operators/lvm/lvm_operator.go (hunk 1 above)
Comment (eifrach) [reply internal/operators/lvm/lvm_operator.go]: changed

File: internal/operators/manager.go (hunk 2 above)
Comment (eifrach) [reply internal/operators/manager.go]: changed

**Comment by eifrach** [comment]
/ test edge-e2e-ai-operator-ztp

**Review by gamli75 — APPROVED** [review]

**Pull request merged**